packages/tpetra/core/src/Tpetra_Details_DistributorActor.hpp

```diff
@@ -227,6 +232,74 @@ void DistributorActor::doPosts(const DistributorPlan& plan,
   // ParameterList set by setParameterList().
   const Details::EDistributorSendType sendType = plan.getSendType();
 
+  //  All-to-all communication layout is quite different from
+  //  point-to-point, so we handle it separately.
+  if (sendType == Details::DISTRIBUTOR_ALLTOALL) {
+#ifdef HAVE_TPETRA_MPI
+    TEUCHOS_TEST_FOR_EXCEPTION(!plan.getIndicesTo().is_null(),
+                               std::runtime_error,
+                               "Send Type=\"Alltoall\" only works for fast-path communication.");
+
+    auto comm = plan.getComm();
+    std::vector<int> sendcounts (comm->getSize(), 0);
```

Comment:
Let's initialize these to `-1` instead of 0 to increase the change of MPI_Alltoallv throwing up if something gets goofed.

Reply:
What if we don't want to send data to some other rank? I'm not sure what MPI will do with a send count of -1, but 0 seems safe to me.